extensions/amp-analytics/0.1/test/test-iframe-transport.js

```diff
@@ -117,6 +119,65 @@ describes.realWin('amp-analytics.iframe-transport', {amp: true}, env => {
     expect(env.win.document.getElementsByTagName('IFRAME')).to.have.lengthOf(0);
   });
 
+  it('creates one PerformanceObserver per vendor type', () => {
+    const createLongTaskObserverSpy = sandbox.spy(
+        IframeTransport.prototype, 'createLongTaskObserver');
+    expect(createLongTaskObserverSpy).to.not.be.called;
+
+    iframeTransport.processCrossDomainIframe(); // Create 2nd frame for 1st vendor
+    expect(createLongTaskObserverSpy).to.not.be.called;
+
+    // Create frame for a new vendor
+    const frameUrl2 = 'https://example.com/test2';
+    new IframeTransport(env.ampdoc.win, 'some_other_vendor_type',
+        {iframe: frameUrl2}, frameUrl2 + '-3');
+    expect(createLongTaskObserverSpy).to.be.called;
+  });
+
+
+  it('logs poor performance of vendor iframe', () => {
+    // Want to change srcdoc before appending
+    const frameUrl2 = 'https://example.com/test2';
+    sandbox.stub(env.ampdoc.win.document.body, 'appendChild');
+    const iframeTransport2 = new IframeTransport(env.ampdoc.win,
+        'some_other_vendor_type', {iframe: frameUrl2}, frameUrl2 + '-3');
+    sandbox.restore();
+    // Clear frame URL, because attrib.containerSrc=='' in test scenario
+    iframeTransport2.frameUrl_ = '';
+    const warnSpy = sandbox.spy(user(), 'warn');
+    const frame = IframeTransport.getFrameData('some_other_vendor_type').frame;
+    frame.srcdoc = '<html><head><script>' +
+        'function busyWait(count, duration, cb) {\n' +
+        '  if (count) {\n' +
+        '    var d = new Date();\n' +
+        '    var d2 = null;\n' +
+        '    do {\n' +
+        '      d2 = new Date();\n' +
+        '    } while (d2-d < duration);\n' + // Note the semicolon!
+        '    setTimeout(function() { busyWait(count-1, duration, cb); },0);\n' +
+        '  } else {\n' +
+        '    cb();\n' +
+        '  }\n' +
+        '}\n' +
+        'function begin() {\n' +
+        '  busyWait(5, 200, function() {\n' +
+        '    window.parent.postMessage("doneSleeping", "*");\n' +
+        '  });\n' +
+        '}' +
+        '</script></head>' +
+        '<body onload="javascript:begin()">Non-Performant Fake Iframe</body>' +
+        '</html>';
+    frame.setAttribute('style', '');
+    env.ampdoc.win.document.body.appendChild(frame);
+    return new Promise((resolve,unused) => {
+      expectPostMessage(frame.contentWindow, env.ampdoc.win, 'doneSleeping')
+          .then(() => {
+            expect(warnSpy).to.be.called;
```

Comment:
is it possible to match the warning msg here. We can skip duration and numLongTasks if they are inconsistent.

Reply:
Done